concepts/observability-configurable-logging/08-revised-api.md

```diff
@@ -0,0 +1,298 @@
+# Revised API
+
+## Motivation For Revise
+
+The basic idea outlined in [the concept for the custom operator](05-custom-fluentbit-operator.md) and [the concept for the servicecatalog integration](06-servicecatalog-integration.md) was to have
+1. a general layer supporting native fluent-bit config snippets. Out of that an overall fluent-bit config gets generated, validated and injected to the damonset
+2. a second layer on top, bringing the convinience of defining pipelines without knowing the underlaying fluent-bit concepts.
+
+The MVP of the overall feature was based on releasing a first version of the 1. native layer. While doing several problems were discovered while doing analysis of several aspects. The These problems adjusted the concepts slightley step by step without re-evaluating the overall picture once again. As a result more and more abstractions/oppinions were build into the first layer, making the layers not differentiated enough anymore. The flexibility on the first layer got lost while the second layer was not fitting anymore fully on top.
+
+## Considerations
+
+### Tag per pipeline mandatory
+
+Different pipelines will require different filters. By default there is only one tag like `kube.*` resulting from the central tail plugin. Every `LogPipeline` will require a dedicated `rewrite_tag` filter in order to apply some filtering, otherwise you will get indeterministic effects (the order of the filters is important).
+-> Every pipeline needs a rewrite_tag introducing a custom tag per pipeline. To generate filters later out of a `LogPreset`, the tag name needs to be known
+
+### Filesystem buffer mandatory
+
+We realized that using in-memory buffers will not decouple the pipelines well enough. If one output is down, the other output will stop working as well. It requires to enable a filesystem buffer for pipeline including size management to assure that the node filesystem gets not flooded
```

Comment:
```suggestion
We realized that using in-memory buffers will not decouple the pipelines well enough. If one output is down, the other output will stop working as well. To assure that the node filesystem is not flooded, there must be an enabled filesystem buffer for each pipeline, including size management.
```
Just guessing the "each".
We could probably drop the word "enabled".

Reply:
Semantic was wrong, re-phrased your suggestion to have it easy to merge